lib/horizon/operation.ex

```diff
@@ -112,27 +112,29 @@ defmodule Stellar.Horizon.Operation do
-  defp operation_type("create_account"), do: %CreateAccount{}
-  defp operation_type("payment"), do: %Payment{}
-  defp operation_type("path_payment_strict_receive"), do: %PathPaymentStrictReceive{}
-  defp operation_type("path_payment_strict_send"), do: %PathPaymentStrictSend{}
-  defp operation_type("manage_sell_offer"), do: %ManageSellOffer{}
-  defp operation_type("manage_buy_offer"), do: %ManageBuyOffer{}
-  defp operation_type("create_passive_sell_offer"), do: %CreatePassiveSellOffer{}
-  defp operation_type("set_options"), do: %SetOptions{}
-  defp operation_type("change_trust"), do: %ChangeTrust{}
-  defp operation_type("allow_trust"), do: %AllowTrust{}
-  defp operation_type("account_merge"), do: %AccountMerge{}
-  defp operation_type("manage_data"), do: %ManageData{}
-  defp operation_type("bump_sequence"), do: %BumpSequence{}
-  defp operation_type("create_claimable_balance"), do: %CreateClaimableBalance{}
-  defp operation_type("claim_claimable_balance"), do: %ClaimClaimableBalance{}
-  defp operation_type("begin_sponsoring_future_reserves"), do: %BeginSponsoringFutureReserves{}
-  defp operation_type("end_sponsoring_future_reserves"), do: %EndSponsoringFutureReserves{}
-  defp operation_type("revoke_sponsorship"), do: %RevokeSponsorship{}
-  defp operation_type("liquidity_pool_deposit"), do: %LiquidityPoolDeposit{}
-  defp operation_type("liquidity_pool_withdraw"), do: %LiquidityPoolWithdraw{}
+  defp operation_type_mapping(type),
+    do: Keyword.merge(@mapping, body: {:struct, operation_type(type)})
+
+  @spec operation_type(type :: String.t()) :: module()
+  defp operation_type("create_account"), do: CreateAccount
+  defp operation_type("payment"), do: Payment
+  defp operation_type("path_payment_strict_receive"), do: PathPaymentStrictReceive
+  defp operation_type("path_payment_strict_send"), do: PathPaymentStrictSend
+  defp operation_type("manage_sell_offer"), do: ManageSellOffer
+  defp operation_type("manage_buy_offer"), do: ManageBuyOffer
+  defp operation_type("create_passive_sell_offer"), do: CreatePassiveSellOffer
+  defp operation_type("set_options"), do: SetOptions
+  defp operation_type("change_trust"), do: ChangeTrust
+  defp operation_type("allow_trust"), do: AllowTrust
+  defp operation_type("account_merge"), do: AccountMerge
+  defp operation_type("manage_data"), do: ManageData
+  defp operation_type("bump_sequence"), do: BumpSequence
+  defp operation_type("create_claimable_balance"), do: CreateClaimableBalance
+  defp operation_type("claim_claimable_balance"), do: ClaimClaimableBalance
+  defp operation_type("begin_sponsoring_future_reserves"), do: BeginSponsoringFutureReserves
+  defp operation_type("end_sponsoring_future_reserves"), do: EndSponsoringFutureReserves
+  defp operation_type("revoke_sponsorship"), do: RevokeSponsorship
+  defp operation_type("liquidity_pool_deposit"), do: LiquidityPoolDeposit
+  defp operation_type("liquidity_pool_withdraw"), do: LiquidityPoolWithdraw
```

Comment:
Did you considered `Macro.camelize/1`?

Reply:
@luishurtado I do. 

After using `Macro.camelize/1` a `String.to_atom/1` will be required since we're looking for a module name (atom). The operation type is an arbitrary value that comes from the Horizon API and IMO performing dynamic string to atom conversions can drive us into memory leaks since atoms are not garbage collected.

We can use `Macro.camelize/1` to reduce function clauses but a guard to match the specific strings will be required anyway. I opted to keep it as a pattern matching